Assets/Scripts/PlayerMovement.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Cinemachine;
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private float moveSpeed;
+    [SerializeField] private float jumpSpeed;
+
+    private float _distanceFromEarth = 0.5f;
+    private float _xVector;
+    private bool _moveable;
+    private Rigidbody2D _rigidbody2D;
+    private Animator _animator;
+    private CinemachineVirtualCamera _cvm;
+
+    private void OnEnable()
+    {
+        Timer.Instance.onFinishCountDown += EnableMoving;
+    }
+
+    private void OnDisable()
+    {
+        Timer.Instance.onFinishCountDown -= EnableMoving;
+    }
+
+    private void Awake()
+    {
+        _moveable = false;
+    }
+
+    private void Start()
+    {
+        _rigidbody2D = GetComponent<Rigidbody2D>();
+        _animator = GetComponent<Animator>();
+    }
+
+    private void Update()
+    {
+        Run();
+        Jump();
+        HandleAnimation();
+    }
+
+    private void Run()
+    {
+        if (_animator.GetBool("IsAlive") && _moveable)
+        {
+            _xVector = Input.GetAxisRaw("Horizontal");
+            transform.position += (Vector3)new Vector2(_xVector * moveSpeed * Time.deltaTime, 0);
+            Flip();
+        }
+    }
+
+    private void Jump()
+    {
+        if (Input.GetKeyDown(KeyCode.Space) && CheckDistanceFromEarth() && _moveable)
+        {
+            _rigidbody2D.AddForce(transform.up * jumpSpeed, ForceMode2D.Impulse);
+        }
+    }
+
+    private void Flip()
+    {
+        if (_xVector != 0)
+        {
+            transform.localScale = (Vector3)new Vector2(_xVector, 1);
+        }
+    }
+
+    private void HandleAnimation()
+    {
+        if (_xVector != 0 && !_animator.GetBool("IsJump"))
+        {
+            _animator.SetFloat("Speed", 1);
+        }
+        else
+        {
+            _animator.SetFloat("Speed", 0);
+        }
+
+        if (!CheckDistanceFromEarth())
+        {
+            _animator.SetBool("IsJump", true);
+        }
+        else
+        {
+            _animator.SetBool("IsJump", false);
+        }
+
+    }
+
+    private bool CheckDistanceFromEarth()
```

Comment:
rename the function to something that should return a boolean, something like `IsOnGround()`